src/universal/components/ThirdPartyAuthButton/ThirdPartyAuthButton.js

```diff
@@ -4,25 +4,29 @@
  * @flow
  */
 
+import type {ThirdPartyAuthProvider} from 'universal/types/auth';
+
 import React from 'react';
 import styled from 'react-emotion';
 
 import Button from 'universal/components/Button/Button';
 
 type Props = {
+  action: string, // E.g. "sign in" or "sign up"
+  disabled?: boolean,
   handleClick: () => void,
-  provider: {
-    iconName: string,
-    displayName: string
-  }
+  provider: ThirdPartyAuthProvider
 };
 
 const ButtonContainer = styled('div')({
   paddingTop: '.5rem'
 });
 
-export default ({provider, handleClick}: Props) => {
-  const label = `Sign in with ${provider.displayName}`;
+const capitalize = (str: string): string =>
```

Comment:
capitalization functions always end up burning me. Since the `action` string is only used as a label, why not just capitalize it before passing it in? if you'd like to decouple the action from the label, I propose a lookup table & making those actions constants & the `action` prop an enum.

Reply:
Yeah fair enough!